### PR TITLE
System.Linq removing duplicated code

### PR DESCRIPTION
**Pull request by Gerko32**
Removing duplicated code by defining helper functions and redefining some methods respect to others.

Performance is not affected.

### CODE REVIEW

File: src/System.Linq/src/System/Linq/Contains.cs [hunk 1]
```diff
@@ -26,20 +26,9 @@ public static bool Contains<TSource>(this IEnumerable<TSource> source, TSource v
...
-            {
-                if (comparer.Equals(element, value))
-                {
-                    return true;
-                }
-            }
+            EnumerableHelpers.ThrowIfNull(source, nameof(source));
 
-            return false;
+            return source.Any(element => comparer.Equals(element, value));
```
Comment (jamesqo) [inline src/System.Linq/src/System/Linq/Contains.cs]: This doesn't look too good... it's adding an additional closure where there wasn't one before.

File: src/System.Linq/src/System/Linq/Aggregate.cs [hunk 2]
```diff
@@ -39,41 +32,14 @@ public static TSource Aggregate<TSource>(this IEnumerable<TSource> source, Func<
...
-            }
-
-            TAccumulate result = seed;
-            foreach (TSource element in source)
-            {
-                result = func(result, element);
-            }
-
-            return result;
+            return source.Aggregate(seed, func, result => result);
```
Comment (jamesqo) [inline src/System.Linq/src/System/Linq/Aggregate.cs]: This is adding an additional delegate invocation that wasn't there before.

Comment (Gerko32) [inline src/System.Linq/src/System/Linq/Aggregate.cs]: True, but is removing duplicated code. it makes it more maintainable. Also this extra delegate is called only once per query independent of size of the array. So, the complexity only increased by O(1)

File: src/System.Linq/src/System/Linq/Contains.cs (hunk 1 above)
Comment (Gerko32) [inline src/System.Linq/src/System/Linq/Contains.cs]: Through, but is removing a lot of duplicated code dough

Comment (stephentoub) [inline src/System.Linq/src/System/Linq/Contains.cs]: @jamesqo is correct.  This is not a change we can take.  It has a very measurable and negative impact on perf.

I appreciate your contribution, and I agree that removing duplication is in general a good thing, but not when it comes at the expense of performance.  Even the changes to replace the if/throw statements can hurt performance, as at least with the JIT today those helper calls will not be inlined and we'll be adding one or more additional function calls to each operation; for some operations on some inputs, that may not matter, but for others, it very well could.  And without proof that it won't have a negative impact, we'd prefer to keep the slightly larger code.

**Comment by stephentoub** [comment]
@Gerko32, thanks again for the interest in contributing.  Per the comments, I'm going to close this PR.

**Pull request closed without merging**